azkaban-web-server/package.json

```diff
@@ -3,18 +3,26 @@
   "name": "azkaban",
   "description": "Azkaban 3 [![Build Status](http://img.shields.io/travis/azkaban/azkaban.svg?style=flat)](https://travis-ci.org/azkaban/azkaban) ========",
   "dependencies": {
-    "chai": "3.5.0",
     "later": "1.2.0",
-    "mocha": "5.0.4",
     "moment": "2.21.0",
     "moment-timezone": "0.5.14",
     "eonasdan-bootstrap-datetimepicker": "4.17.47",
     "rewire": "2.5.2",
     "underscore": "1.8.3",
     "ycssmin": "1.0.1"
   },
+  "devDependencies": {
+    "chai": "3.5.0",
+    "chai-http": "4.0.0",
+    "mocha": "5.0.4"
+  },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "pretest": "npm run startSoloServer",
+    "test": "mocha --recursive 'src/web/js/azkaban/test/' || npm run stop_n_exit",
```

Comment:
test_running_server.js and package.json are used by `npm test` to start a azkaban-solo instance and confirms the web page renders.  

test.js is used by `gradle test` and only tests JavaScript functions return expected results.  A azkaban-solo instance is not started by `gradle test`.

Reply:
I guess I am still not sure how `npm test` knows to run `test_running_server.js` but not `test.js` when its parameter only specifies the (entire?) directory. Will it not run both? Or maybe it will, and it's ok?

Reply:
By default Mocha by default looks for "test/*js", [documentation here](https://mochajs.org/#the-test-directory).  We are overriding default behavior by passing a directory path as an argument to mocha. 

To answer your question, `NPM test` will run tests from both files.  I think this is okay because the few seconds spent testing components is much smaller then waiting for running tests to complete.